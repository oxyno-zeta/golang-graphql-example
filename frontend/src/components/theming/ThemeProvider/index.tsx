import ThemeProvider, { Props as ThemeProviderProps } from './ThemeProvider';

export default ThemeProvider;
export type { ThemeProviderProps };
