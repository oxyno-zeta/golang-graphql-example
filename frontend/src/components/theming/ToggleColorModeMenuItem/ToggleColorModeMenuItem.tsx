import React, { useContext } from 'react';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import { useTranslation } from 'react-i18next';
import Typography from '@mui/material/Typography';
import { useTheme } from '@mui/material/styles';
import SvgIcon from '@mui/material/SvgIcon';
import { mdiBrightness2, mdiBrightness7 } from '@mdi/js';
import { PaletteMode } from '@mui/material';
import ColorModeContext from '../../../contexts/ColorModeContext';

interface Props {
  titleStyle?: React.CSSProperties;
}

const defaultProps = {
  titleStyle: { fontSize: 13, marginBottom: '2px' },
};

function ToggleColorModeMenuItem({ titleStyle }: Props) {
  // Setup translate
  const { t } = useTranslation();
  // Get theme
  const theme = useTheme();
  // Get color mode context
  const colorMode = useContext(ColorModeContext);

  // Expand
  const { setColorMode } = colorMode;

  return (
    <>
      <Typography style={titleStyle}>{t('common.themeTitle')}</Typography>
      <ToggleButtonGroup
        fullWidth
        size="small"
        value={theme.palette.mode}
        exclusive
        onChange={(event, value) => {
          setColorMode(value as PaletteMode);
        }}
      >
        <ToggleButton value="dark">
          <SvgIcon sx={{ marginRight: '5px' }}>
            <path d={mdiBrightness2} />
          </SvgIcon>{' '}
          {t('common.darkThemeSelector')}
        </ToggleButton>
        <ToggleButton value="light">
          <SvgIcon sx={{ marginRight: '5px' }}>
            <path d={mdiBrightness7} />
          </SvgIcon>{' '}
          {t('common.lightThemeSelector')}
        </ToggleButton>
      </ToggleButtonGroup>
    </>
  );
}

ToggleColorModeMenuItem.defaultProps = defaultProps;

export default ToggleColorModeMenuItem;
