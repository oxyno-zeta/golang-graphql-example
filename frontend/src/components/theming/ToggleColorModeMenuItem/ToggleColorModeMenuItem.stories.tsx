import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import CssBaseline from '@mui/material/CssBaseline';
import Typography from '@mui/material/Typography';
import { useTheme } from '@mui/material/styles';
import ThemeProvider from '~components/theming/ThemeProvider';
import ToggleColorModeMenuItem, { Props } from './ToggleColorModeMenuItem';

export default {
  title: 'Components/theming/ToggleColorModeMenuItem',
  component: ToggleColorModeMenuItem,
} as Meta<typeof ToggleColorModeMenuItem>;

function Info() {
  const theme = useTheme();
  return <Typography>Theme mode:{theme.palette.mode}</Typography>;
}

export const Playbook: StoryFn<typeof ToggleColorModeMenuItem> = function C(args: Props) {
  return (
    <ThemeProvider themeOptions={{}}>
      <CssBaseline />
      <ToggleColorModeMenuItem {...args} />
      <Info />
    </ThemeProvider>
  );
};
