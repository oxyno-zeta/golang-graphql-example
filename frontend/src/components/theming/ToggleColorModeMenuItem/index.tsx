import ToggleColorModeMenuItem, { Props as ToggleColorModeMenuItemProps } from './ToggleColorModeMenuItem';

export default ToggleColorModeMenuItem;
export type { ToggleColorModeMenuItemProps };
