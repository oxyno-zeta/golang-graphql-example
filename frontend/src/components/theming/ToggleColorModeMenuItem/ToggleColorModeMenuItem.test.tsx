import React from 'react';
import { render, fireEvent } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import ThemeProvider from '~components/theming/ThemeProvider';

import ToggleColorModeMenuItem from './ToggleColorModeMenuItem';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

let matchMedia: (q: string) => MediaQueryList;

describe('theming/ToggleColorModeMenuItem', () => {
  beforeAll(() => {
    matchMedia = window.matchMedia;
  });

  afterAll(() => {
    window.matchMedia = matchMedia;
  });

  it('should display the light theme as selected and switch to dark', async () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      if (query === '(prefers-color-scheme: dark)') {
        return {
          matches: false,
          media: query,
          onchange: null,
          addListener: jest.fn(), // deprecated
          removeListener: jest.fn(), // deprecated
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
          dispatchEvent: jest.fn(),
        };
      }

      return matchMedia(query);
    });

    const { container } = render(
      <ThemeProvider themeOptions={{}}>
        <ToggleColorModeMenuItem />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();

    expect(container).toHaveTextContent('common.themeTitle');
    expect(container).toHaveTextContent('common.darkThemeSelector');
    expect(container).toHaveTextContent('common.lightThemeSelector');

    const allButtons = container.querySelectorAll('button');

    expect(allButtons[0]).toHaveAttribute('value', 'dark');
    expect(allButtons[1]).toHaveAttribute('value', 'light');

    expect(allButtons[1]).toHaveClass('Mui-selected');

    expect(fireEvent.click(allButtons[0])).toBeTruthy();

    expect(allButtons[0]).toHaveClass('Mui-selected');

    expect(container).toMatchSnapshot();
  });

  it('should display the dark theme as selected and switch to light', async () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      if (query === '(prefers-color-scheme: dark)') {
        return {
          matches: true,
          media: query,
          onchange: null,
          addListener: jest.fn(), // deprecated
          removeListener: jest.fn(), // deprecated
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
          dispatchEvent: jest.fn(),
        };
      }

      return matchMedia(query);
    });

    const { container } = render(
      <ThemeProvider themeOptions={{}}>
        <ToggleColorModeMenuItem />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();

    expect(container).toHaveTextContent('common.themeTitle');
    expect(container).toHaveTextContent('common.darkThemeSelector');
    expect(container).toHaveTextContent('common.lightThemeSelector');

    const allButtons = container.querySelectorAll('button');

    expect(allButtons[0]).toHaveAttribute('value', 'dark');
    expect(allButtons[1]).toHaveAttribute('value', 'light');

    expect(allButtons[0]).toHaveClass('Mui-selected');

    expect(fireEvent.click(allButtons[1])).toBeTruthy();

    expect(allButtons[1]).toHaveClass('Mui-selected');

    expect(container).toMatchSnapshot();
  });
});
