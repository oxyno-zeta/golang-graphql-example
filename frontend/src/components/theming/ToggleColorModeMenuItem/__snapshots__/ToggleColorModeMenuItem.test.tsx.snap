// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theming/ToggleColorModeMenuItem should display the dark theme as selected and switch to light 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 7px;
  border: 1px solid rgba(255, 255, 255, 0.12);
  color: #fff;
  width: 100%;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
  border: 1px solid rgba(255, 255, 255, 0.12);
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(255, 255, 255, 0.08);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-selected {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.16);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(255, 255, 255, 0.24);
}

@media (hover: none) {
  .emotion-2.Mui-selected:hover {
    background-color: rgba(255, 255, 255, 0.16);
  }
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
    style="font-size: 13px; margin-bottom: 2px;"
  >
    common.themeTitle
  </p>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-1"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="dark"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,2C8.18,2 6.47,2.5 5,3.35C8,5.08 10,8.3 10,12C10,15.7 8,18.92 5,20.65C6.47,21.5 8.18,22 10,22A10,10 0 0,0 20,12A10,10 0 0,0 10,2Z"
        />
      </svg>
       
      common.darkThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="light"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,8A4,4 0 0,0 8,12A4,4 0 0,0 12,16A4,4 0 0,0 16,12A4,4 0 0,0 12,8M12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31L23.31,12L20,8.69Z"
        />
      </svg>
       
      common.lightThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</div>
`;

exports[`theming/ToggleColorModeMenuItem should display the dark theme as selected and switch to light 2`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 7px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-selected {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-2.Mui-selected:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
    style="font-size: 13px; margin-bottom: 2px;"
  >
    common.themeTitle
  </p>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-1"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="dark"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,2C8.18,2 6.47,2.5 5,3.35C8,5.08 10,8.3 10,12C10,15.7 8,18.92 5,20.65C6.47,21.5 8.18,22 10,22A10,10 0 0,0 20,12A10,10 0 0,0 10,2Z"
        />
      </svg>
       
      common.darkThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="light"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,8A4,4 0 0,0 8,12A4,4 0 0,0 12,16A4,4 0 0,0 16,12A4,4 0 0,0 12,8M12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31L23.31,12L20,8.69Z"
        />
      </svg>
       
      common.lightThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</div>
`;

exports[`theming/ToggleColorModeMenuItem should display the light theme as selected and switch to dark 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 7px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-selected {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-2.Mui-selected:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
    style="font-size: 13px; margin-bottom: 2px;"
  >
    common.themeTitle
  </p>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-1"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="dark"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,2C8.18,2 6.47,2.5 5,3.35C8,5.08 10,8.3 10,12C10,15.7 8,18.92 5,20.65C6.47,21.5 8.18,22 10,22A10,10 0 0,0 20,12A10,10 0 0,0 10,2Z"
        />
      </svg>
       
      common.darkThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="light"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,8A4,4 0 0,0 8,12A4,4 0 0,0 12,16A4,4 0 0,0 16,12A4,4 0 0,0 12,8M12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31L23.31,12L20,8.69Z"
        />
      </svg>
       
      common.lightThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</div>
`;

exports[`theming/ToggleColorModeMenuItem should display the light theme as selected and switch to dark 2`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-1 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 7px;
  border: 1px solid rgba(255, 255, 255, 0.12);
  color: #fff;
  width: 100%;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
  border: 1px solid rgba(255, 255, 255, 0.12);
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(255, 255, 255, 0.08);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-selected {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.16);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(255, 255, 255, 0.24);
}

@media (hover: none) {
  .emotion-2.Mui-selected:hover {
    background-color: rgba(255, 255, 255, 0.16);
  }
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
    style="font-size: 13px; margin-bottom: 2px;"
  >
    common.themeTitle
  </p>
  <div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-1"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="dark"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10,2C8.18,2 6.47,2.5 5,3.35C8,5.08 10,8.3 10,12C10,15.7 8,18.92 5,20.65C6.47,21.5 8.18,22 10,22A10,10 0 0,0 20,12A10,10 0 0,0 10,2Z"
        />
      </svg>
       
      common.darkThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-2"
      tabindex="0"
      type="button"
      value="light"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,8A4,4 0 0,0 8,12A4,4 0 0,0 12,16A4,4 0 0,0 16,12A4,4 0 0,0 12,8M12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6A6,6 0 0,1 18,12A6,6 0 0,1 12,18M20,8.69V4H15.31L12,0.69L8.69,4H4V8.69L0.69,12L4,15.31V20H8.69L12,23.31L15.31,20H20V15.31L23.31,12L20,8.69Z"
        />
      </svg>
       
      common.lightThemeSelector
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</div>
`;
